namespace/namespace.go

```diff
@@ -7,8 +7,7 @@ import (
 )
 
 type Namespace struct {
-	Version uint8
-	ID      []byte
+	data []byte
 }
 
 // New returns a new namespace with the provided version and id.
```

Comment:
I recommend having a versioned constructor. This is helpful for devs to construct the blob with a namespace. It's usually unclear where to get the version constant and which one to use, while a constructor for every version(along general one) simplifies that. Considering that new versions aren't coming out often, adding a new constructor for each would have add much overhead.

Reply:
Besides, each constructor may address the quirks of a namespace. For example, V0 requires first N zero bytes for the ID, and expecting the user to know that to construct the correct ID is ridiculous. The specialized constructor can expect exactly the needed amount of bytes that the user controls.

Reply:
There is already `NewV0`

Reply:
Missed that.

Reply:
Can we make sure NewV0 allocates once as well? Currently, it does that twice in itself and in the `newNamespace`?

Reply:
The main reason is that this is worse then the node's status quo and migrating over it is meh

Reply:
Have a look now